ya# (svelte/naming-convention-in-kit-routes)

> description

## :book: Rule Details

This rule reports ???.

<ESLintCodeBlock>

<!--eslint-skip-->

```svelte
<script>
  /* eslint svelte/naming-convention-in-kit-routes: "error" */
</script>

<!-- ✓ GOOD -->

<!-- ✗ BAD -->
```

</ESLintCodeBlock>

## :wrench: Options

```json
{
  "svelte/naming-convention-in-kit-routes": ["error", {}]
}
```

-

## :books: Further Reading

-
