# (svelte/no-export-load-in-svelte-module-in-kit-pages)

> description

## :book: Rule Details

This rule reports ???.

<ESLintCodeBlock>

<!--eslint-skip-->

```svelte
<script>
  /* eslint svelte/no-export-load-in-svelte-module-in-kit-pages: "error" */
</script>

<!-- ✓ GOOD -->

<!-- ✗ BAD -->
```

</ESLintCodeBlock>

## :wrench: Options

```json
{
  "svelte/no-export-load-in-svelte-module-in-kit-pages": ["error", {}]
}
```

-

## :books: Further Reading

-
